testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt

```diff
@@ -91,11 +94,24 @@ class MiniAppDisplayActivity : BaseActivity() {
                     callback: (isGranted: Boolean) -> Unit
                 ) {
                     miniappPermissionCallback = callback
-                    ActivityCompat.requestPermissions(
-                        this@MiniAppDisplayActivity,
-                        AppPermission.getPermissionRequest(miniAppPermissionType),
-                        AppPermission.getRequestCode(miniAppPermissionType)
-                    )
+
+                    when (miniAppPermissionType.platform) {
+                        MiniAppPermissionPlatform.Android.name -> {
+                            // requesting Android specific permission
+                            ActivityCompat.requestPermissions(
+                                this@MiniAppDisplayActivity,
+                                AppPermission.getPermissionRequest(miniAppPermissionType),
+                                AppPermission.getRequestCode(miniAppPermissionType)
+                            )
+                        }
+                        MiniAppPermissionPlatform.MiniApp.name -> {
+                            // requesting MiniApp SDK specific permission
+                            MiniApp.requestPermission(
+                                this@MiniAppDisplayActivity,
+                                AppPermission.getSinglePermissionRequest(miniAppPermissionType)
```

Comment:
For the data related requests, the Mini App will want to send an array of permissions which the Demo App should show all at once in a single dialog. 

Currently, the SDK only supports requesting one permission at a time, so maybe we need to add an interface `MiniAppMessageBridge#requestCustomPermissions`?